OpenFOAM/bike_OpenFOAM8.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+if [[ $EESSI_CVMFS_REPO == "/cvmfs/software.eessi.io" ]] && [[ $EESSI_VERSION == "2023.06" ]]; then module load OpenFOAM/10-foss-2023a
+elif [[ $EESSI_CVMFS_REPO == "/cvmfs/pilot.eessi-hpc.org" ]] && [[ $EESSI_PILOT_VERSION == "2021.12" ]]; then module load OpenFOAM/8-foss-2020a
+elif [[ $EESSI_CVMFS_REPO == "/cvmfs/pilot.eessi-hpc.org" ]] && [[ $EESSI_PILOT_VERSION == "2023.06" ]]; 
+then echo There is no demo for OpenFOAM in "/cvmfs/pilot.eessi-hpc.org/versions/2023.06". Please use the EESSI production repo "/cvmfs/software.eessi.io".;
+exit 1;
+else echo "Don't know which OpenFOAM module to load for ${EESSI_CVMFS_REPO}/versions/${EESSI_VERSION}$EESSI_PILOT_VERSION" >&2; exit 1
+fi
+
+
+which ssh &> /dev/null
+if [ $? -ne 0 ]; then
+    # if ssh is not available, set plm_rsh_agent to empty value to avoid OpenMPI failing over it
+    # that's OK, because this is a single-node run
+    export OMPI_MCA_plm_rsh_agent=''
+fi
+
+source $FOAM_BASH
+
+if [ -z $EBROOTOPENFOAM ]; then
+    echo "ERROR: OpenFOAM module not loaded?" >&2
+    exit 1
+fi
+
+# Allow users to define the WORKDIR externally (for example a shared FS for multinode runs)
+export WORKDIR="${WORKDIR:-/tmp/$USER/$$}"
+echo "WORKDIR: $WORKDIR"
+mkdir -p $WORKDIR
+cd $WORKDIR
+pwd
+
+# motorBike, 2M cells
+BLOCKMESH_DIMENSIONS="100 40 40"
+# motorBike, 150M cells
+#BLOCKMESH_DIMENSIONS="200 80 80"
+
+# X*Y*Z should be equal to total number of available cores (across all nodes)
+X=${X:-4}
+Y=${Y:-2}
+Z=${Z:-1}
+# number of nodes
+NODES=${NODES:-1}
+# total number of cores
+NP=$((X * Y * Z))
+# cores per node
+PPN=$(((NP + NODES -1)/NODES))
+
+CASE_NAME=motorBike
+
+if [ -d $CASE_NAME ]; then
+    echo "$CASE_NAME already exists in $PWD!" >&2
+    exit 1
+fi
+
+cp -r $WM_PROJECT_DIR/tutorials/incompressible/simpleFoam/motorBike $CASE_NAME
+cd $CASE_NAME
+pwd
+
+# generate mesh
+echo "generating mesh..."
+foamDictionary  -entry castellatedMeshControls.maxGlobalCells -set 200000000 system/snappyHexMeshDict
+foamDictionary -entry blocks -set "( hex ( 0 1 2 3 4 5 6 7 ) ( $BLOCKMESH_DIMENSIONS ) simpleGrading ( 1 1 1 ) )" system/blockMeshDict
+foamDictionary -entry numberOfSubdomains -set $NP system/decomposeParDict
+foamDictionary -entry hierarchicalCoeffs.n -set "($X $Y $Z)" system/decomposeParDict
+
+cp $WM_PROJECT_DIR/tutorials/resources/geometry/motorBike.obj.gz constant/triSurface/
+surfaceFeatures 2>&1 | tee log.surfaceFeatures
+blockMesh 2>&1 | tee log.blockMesh
+decomposePar -copyZero 2>&1 | tee log.decomposePar
+mpirun -np $NP -ppn $PPN -hostfile hostlist snappyHexMesh -parallel -overwrite 2>&1 | tee log.snappyHexMesh
```

Comment:
```suggestion
mpirun -np $NP --oversubscribe snappyHexMesh -parallel -overwrite 2>&1 | tee log.snappyHexMesh
```

Reply:
Yes, I have removed this in my latest script which @laraPPr is still testing. Her next commit should make things clear.

Reply:
In Satish script for version 11 it is changed to this `mpirun -np $NP -npernode $PPN snappyHexMesh -parallel -overwrite 2>&1 | tee log.snappyHexMesh`. And that one now works correctly. I'm also less inclined to change this one because it still does something and is only their for the OpenFOAM in the pilot repo.

Reply:
I don't really have a problem with this (since it will soon be lost to history anyway), but the file `hostlist` does not exist as far as I can see, so it really does nothing (or is it just a workaround for the use of `-ppn`?).

Reply:
Actually `-ppn` is deprecated and `-npernode` is to be used instead with `mpirun`. Indeed the hostlist thing didn't make any sense to me so I also removed it as @ocaisa did.